botocore/client.py

```diff
@@ -208,6 +209,10 @@ def _get_client_args(self, service_model, region_name, is_secure,
         self._inject_s3_configuration(
             config_kwargs, scoped_config, client_config)
         self._conditionally_unregister_fix_s3_host(endpoint_url, event_emitter)
+        if service_name == 's3':
+            # We want to make double sure that we don't create this object
```

Comment:
It's already S3 specific, the name of the class includes 'S3' as do the names of the functions above. I don't see how this is any more ugly. If I had to go to any amount of effort to get the name of the service I would agree, but it's readily available. It seems irresponsible to create this object and hold it in memory for the life of the client when it isn't being used for any other service given how easy it is to avoid that.